src/shared/ui/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+
+import useOutsideClick from './useOutsideClick';
+
+import ModalPortal from '../Portal/Portal';
+
+type ModalProps = {
+  isOpen: boolean;
+  closeModal: () => void;
+  modalRef: React.RefObject<HTMLDivElement>;
+  children: React.ReactNode;
+};
+
+export default function Modal({ isOpen, closeModal, modalRef, children }: ModalProps) {
+  useOutsideClick(modalRef, closeModal);
+
+  return (
+    <ModalPortal>
+      <AnimatePresence>
+        {isOpen && (
```

Comment:
자주 사용되는 `Portal`은 따로 해당 컴포넌트 파일 내에서 공통으로 뺄 것 같아요~!

++ 확장성을 위해 `mode`도 `Props`로 설정해두면 좋을 것 같네용
```suggestion
    <PortalWrapper>
      <AnimatePresence mode="wait">{isOpen && children}</AnimatePresence>
    </PortalWrapper>
```